packages/mermaid/src/diagram-api/diagram-orchestration.ts

```diff
@@ -55,25 +55,28 @@ export const addDiagrams = () => {
       return text.toLowerCase().trimStart().startsWith('---');
     }
   );
+  // Ordering of detectors is important. The first one to return true will be used.
```

Comment:
I'd recommend adding a TypeScript function comment to https://github.com/mermaid-js/mermaid/blob/f3b313ec1d670bc426e1e0876fc61f154f577c48/packages/mermaid/src/diagram-api/detectType.ts#L52 too.

Something like:

```typescript
/**
 * Registers lazy-loaded diagrams to Mermaid.
 *
 * The diagram function is loaded asynchronously, so that diagrams are only loaded
 * if the diagram is detected.
 *
 * @remarks
 * Please note that the order of diagram detectors is important.
 * The first detector to return `true` is the diagram that will be loaded
 * and used, so put more specific detectors at the beginning!
 *
 * @param diagrams - Diagrams to lazy load, and their detectors, in order of importance.
 */
```

Reply:
Done